sycl/include/CL/sycl/intel/builtins.hpp

```diff
@@ -0,0 +1,69 @@
+//==------ builtins.hpp - Non-standard SYCL built-in functions -------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+
+#ifdef __SYCL_DEVICE_ONLY__
+#define CONSTANT_AS __attribute__((opencl_constant))
+// Note: __format string is declared in constant address space to be compatible
+// with OpenCL C
+extern int __spirv_ocl_printf(const CONSTANT_AS char *__format, ...)
+    __attribute__((format(printf, 1, 2)));
+#else
+#define CONSTANT_AS
+#endif
+
+namespace cl {
+namespace sycl {
+namespace intel {
+namespace experimental {
+
+// Provides functionality to print data from kernels in a C way. On non-HOST
+// devices this function is directly mapped to printf from OpenCL C - see
+// corresponding specification to get information about format string.
+//
+// On host device, this function should be equivalent to standard printf
+// function from C/C++.
+//
```

Comment:
That means we cannot use vectors in format string, and cannot print
NULL pointers and expect "(null)" string.

Reply:
Do I really need to mention all the differences here? I think something like "please refer to corresponding section of OpenCL C specification to get information about format string differences from C standard rules"

Reply:
Just mention that no additional feature can be used on host or host device.

Reply:
Done